### PR TITLE
Add List Ingress Health for Service

### LINKED ISSUE #349: [FEATURE REQUEST]: List Ingress Health for Service
### Is your feature request related to a problem? Please describe.

No

### Describe the solution you'd like

Add support to GET /v1/health/ingress/ + service https://developer.hashicorp.com/consul/api-docs/health#list-service-instances-for-ingress-gateways-associated-with-a-service
This endpoint returns the service instances providing an [ingress gateway](https://developer.hashicorp.com/consul/docs/connect/gateways/ingress-gateway) for a service in a given datacenter.
This endpoint is supported in Golang client (1.9.17) https://github.com/hashicorp/consul/blob/main/api/health.go#L326

### Describe alternatives you've considered

Nothing for the moment

### Additional context

Nothing for the moment

### PR DESCRIPTION
**Pull request by IvanKolchanov**
Added endpoint and tests for it, had to change Node in Catalog.cs because it lacked a field
Resolves #349

### CODE REVIEW

File: Consul.Test/HealthTest.cs [hunk 1]
```diff
@@ -157,6 +158,54 @@ public async Task Health_Connect()
...
+                                }
+                            }
+                        }
+                    }
+            };
+            await _client.Configuration.ApplyConfig(ingressGatewayConfigEntry);
+
+            var services = await _client.Health.Ingress("foo-ingress", "", false);
+            Assert.Single(services.Response);
+            Assert.Equal("dc1", services.Response[0].Node.Datacenter);
```
Comment (marcin-krystianc) [inline Consul.Test/HealthTest.cs:205]: It is better to assert that the data center is not empty ( to make the test independent from external configuration).

**Review by marcin-krystianc — COMMENTED** [review]
looks, good. Just a few minor comments.

**Comment by IvanKolchanov** [comment]
@marcin-krystianc fixed everything, I think

**Review by marcin-krystianc — APPROVED** [review]
Nice, thank you.

**Pull request merged**